### PR TITLE
Improved efficiency of Update method in TickedQueue

### PR DESCRIPTION
**Pull request by Hodkinson**
The _queue.Remove(item); line in TickedQueue was shifting the array back each time a Tick happened - this changes the internal representation from a List to a SortedSet, to allow O(log n) removals, at the expense of storing an extra long with each TickedQueueItem.

I think I messed up the spacing though, I don't have Visual Studio.

### CODE REVIEW

**Comment by ricardojmendez** [comment]
Hi Hodkinson,

Thanks for the PR, but it's not passing the Unit tests.  Check TickedQueueUnit.TestUnPauseBasic.

**Comment by Hodkinson** [comment]
Sorry about that - when running the tests locally in MonoDevelop they all are green - but when I run that one individually it fails. I'll have a look.

**Comment by Hodkinson** [comment]
Turns out that test was always wrong, it was testing against the wrong value and the timing was messed up which it probably why it was passing before. I've fixed it with the latest commit.

**Comment by ricardojmendez** [comment]
Formidable, thanks. Will review and merge.

File: TickedPriorityQueue/Source/Implementation/TickedQueueItem.cs [hunk 1]
```diff
@@ -45,10 +53,12 @@ public TickedQueueItem(ITicked ticked, DateTime currentTime, bool isLooped = tru
 		{
             if (ticked == null) throw new ArgumentNullException("Missing a valid ITicked reference");
             _ticked = ticked;
+            _creationId = nextCreationId++;
```
Comment (ricardojmendez) [inline TickedPriorityQueue/Source/Implementation/TickedQueueItem.cs:56]: This change is always indexing a TickedQueueItem by _creationId. The _creationId only gets assigned on initialization, and as far as I can see, never again.   We then sort by _creationId;

However, TickedQueue _can_ re-add items on looping, in which case the just-added items should go to the end. Need to verify that this is actually happening, see if the tests are covering the case.

**Review by ricardojmendez — COMMENTED** [review]
Bit rusty on TickedPriorityQueue, since I haven't modified it in a while. Will need to review this further.

**Pull request closed without merging**